src/PrestaShopBundle/Entity/Repository/CmsPageRepository.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Entity\Repository;
+
+use CMS;
+use Language;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\CmsPageSettings;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\Exception\CannotDuplicateCmsPageException;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\Exception\CmsPageNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\CmsPage\ValueObject\CmsPageId;
+use PrestaShop\PrestaShop\Core\Repository\AbstractObjectModelRepository;
+use PrestaShop\PrestaShop\Core\Util\String\StringModifierInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class CmsPageRepository extends AbstractObjectModelRepository
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var StringModifierInterface
+     */
+    private $stringModifier;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param StringModifierInterface $stringModifier
+     */
+    public function __construct(
+      TranslatorInterface $translator,
+      StringModifierInterface $stringModifier
+  ) {
+      $this->translator = $translator;
+      $this->stringModifier = $stringModifier;
+  }
+
+    /**
+     * Duplicates a CMS page
+     *
+     * @param CmsPageId $cmsPageId
+     *
+     * @return CmsPageId
+     *
+     * @throws CmsPageNotFoundException
+     * @throws CannotDuplicateCmsPageException
+     */
+    public function duplicate(CmsPageId $cmsPageId): CmsPageId
```

Comment:
I think we will need to do it as in ProductRepository. So this method should accept whole CmsPage object model and only handle the unsetting of id and saving the obj model, while leaving the rest of the logic for dedicated CmsDuplicator service (yeah, more code, but also more SOLID).
Because the "getNewPageTitles" logic does not belong to the repository pattern, so it should stay outside.

Reply:
Done ✅